### PR TITLE
[GPU] Fixed deserialization logic for dynamic batch

### PR DESCRIPTION
**Pull request by e-ddykim**
### Details:
 - This PR fixes deserialization logic for dynamic batch

### Tickets:
 - 109497

### CODE REVIEW

**Review by ahnyoung-paul — APPROVED** [review]

File: src/plugins/intel_gpu/include/intel_gpu/graph/serialization/binary_buffer.hpp [hunk 1]
```diff
@@ -36,32 +37,43 @@ class BinaryOutputBuffer : public OutputBuffer<BinaryOutputBuffer> {
...
+        return _const_data_map[net_id][prim_id];
     }
 
-    std::streampos tellg() { return stream.tellg(); }
-    void seekg(std::streampos pos) { stream.seekg(pos); }
+    std::streampos tellg() { return _stream.tellg(); }
+    void seekg(std::streampos pos) { _stream.seekg(pos); }
+
+    void new_network_added() { _num_networks += 1; }
+    int get_num_networks() { return _num_networks; }
```
Comment (andrew-k-park) [inline src/plugins/intel_gpu/include/intel_gpu/graph/serialization/binary_buffer.hpp:66]: nit: It seems better to specify const keyword for "read-only" function.
```suggestion
    int get_num_networks() const { return _num_networks; }
```

File: src/plugins/intel_gpu/include/intel_gpu/graph/serialization/binary_buffer.hpp [hunk 2]
```diff
@@ -36,32 +37,43 @@ class BinaryOutputBuffer : public OutputBuffer<BinaryOutputBuffer> {
...
+
+    void set_stream_id(uint16_t stream_id) { _stream_id = stream_id; }
+    uint16_t get_stream_id() { return _stream_id; }
```
Comment (andrew-k-park) [inline src/plugins/intel_gpu/include/intel_gpu/graph/serialization/binary_buffer.hpp:69]: nit: same as above

**Review by andrew-k-park — APPROVED** [review]
Overall looks good to me with minor comments

File: src/plugins/intel_gpu/include/intel_gpu/graph/serialization/binary_buffer.hpp (hunk 1 above)
Comment (e-ddykim) [reply src/plugins/intel_gpu/include/intel_gpu/graph/serialization/binary_buffer.hpp]: I added "const" as you commented. Thank you.

File: src/plugins/intel_gpu/src/graph/data.cpp [hunk 3]
```diff
@@ -103,7 +103,7 @@ void data_inst::load(BinaryInputBuffer& ib) {
             _outputs[0]->copy_from(get_network().get_stream(), _buf.data());
         }
 
-        ib.addConstData(id(), _outputs[0]);
+        ib.addConstData(get_network_id(), id(), _outputs[0]);
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/graph/data.cpp:106]: Does it always copy constant for all networks not created for dynamic batch? i.e., networks created for multiple streams, not for the dynamic batches..

Comment (e-ddykim) [reply src/plugins/intel_gpu/src/graph/data.cpp:106]: For the case of multiple streams, const data is shared among them by L90.

**Review by yeonbok — APPROVED** [review]

**Pull request merged**